pangeo_forge_recipes/aggregation.py

```diff
@@ -23,11 +23,17 @@ class XarraySchema(TypedDict):
 
 
 def dataset_to_schema(ds: xr.Dataset) -> XarraySchema:
-    """Convert the output of `dataset.to_dict(data=False)` to a schema
+    """Convert the output of `dataset.to_dict(data=False, encoding=True)` to a schema
     (Basically justs adds chunks, which is not part of the Xarray ouput).
     """
 
-    d = ds.to_dict(data=False)
+    # Remove redundant encoding options
+    for v in ds.variables:
+        for option in ["_FillValue", "source"]:
```

Comment:
Could you explain the rationale for special casing these two options?

Reply:
I excluded these as they were causing certain test failures where expected schemas were compared with actual. E.g. any combiner tests using `has_correct_schema()`:

https://github.com/pangeo-forge/pangeo-forge-recipes/blob/beam-refactor/tests/test_combiners.py#L98-L102

```python
def has_correct_schema(expected_schema):
    def _check_results(actual):
        assert len(actual) == 1
        schema = actual[0]
        assert schema == expected_schema
```

The `source` will be unique to each original source data product, and the `_FillValue` appeared to be added automatically (I can't recall the specific issue with the latter though).

Reply:
I checked the latter again, and when `_FillValue` is retained, it's being set to `nan` in `expected_schema` (as generated by the original `ds.to_dict(data=False, encoding=True)`, but only for the `lat` and `lon` coords. However, the `actual` schema doesn't contain `_FillValue` for `lat`/`lon`, and the assert fails.